### DISCUSSION TITLE
Support `/pin` Twitch command

### DISCUSSION THREAD

**Discussion by MattIPv4** [Chatterino/chatterino2#5165] [Ideas]
### Checklist

- [X] I'm reporting a problem with Chatterino
- [X] I've verified that I'm running **the most** recent nightly build or stable release
- [X] I've looked for my problem on the [wiki](https://wiki.chatterino.com/Help/)
- [X] I've searched the [issues and pull requests](https://github.com/Chatterino/chatterino2/issues?q=) for similar looking reports

### Describe your issue

Running `/pin <message>` results in `Unrecognized command: /pin`

### OS and Chatterino Version

Chatterino 2.4.6 (commit 137cd7a4 modified) built with Qt 6.5.0 Running on macOS 14.3, kernel: 23.3.0

**Comment by Felanbird** [comment]
There's no API for pinning, we can't add it

**Reply by Schokolabbi** [reply]
How about sending /pin {1+} in chat?
https://help.twitch.tv/s/article/chat-commands?language=en_US#AllMods

**Reply by pajlada** [reply]
We can't send that in chat, only first party clients can iirc

**Reply by Mm2PL** [reply]
I wish it was that simple, commands used to be like that but twitch decided to replace them with an api, helix. For us to use we need twitch to take their sweet time to implement an endpoint, this can take a long time especially if the feature is an experiment. And then there's the luck part, if we get unlucky* the required permissions on the endpoint will be set up in a way that we can't reasonably use it.

**Comment by jupjohn** [comment]
Related uservoice (marked as "planned"): https://twitch.uservoice.com/forums/310213-developers/suggestions/45606919-pinned-messages-api

**Comment by ZonianMidian** [comment]
New _beta_ API endpoints:
[Get Pinned Chat Message](https://dev.twitch.tv/docs/api/reference#get-pinned-chat-message)
[Pin Chat Message](https://dev.twitch.tv/docs/api/reference#pin-chat-message)
[Update Pinned Chat Message](https://dev.twitch.tv/docs/api/reference#update-pinned-chat-message)
[Unpin Chat Message](https://dev.twitch.tv/docs/api/reference#unpin-chat-message)

The [Send Chat Message API](https://dev.twitch.tv/docs/api/reference#send-chat-message) endpoint has been updated with a new optional pin parameter to support sending and pinning a message in a single request.

**Reply by jupjohn** [reply]
Worth noting that all of these endpoints are moderator-only, so normal users will not be able to get currently pinned messages. EventSub support is still missing.

I've created #7000 for any feature design thoughts since GH discussions with multiple threads are painful to follow.
